python-sdk/hatchet_sdk/clients/events.py

```diff
@@ -1,37 +1,44 @@
-from ..events_pb2_grpc import EventsServiceStub
-from ..events_pb2 import PushEventRequest, PutLogRequest
-
 import datetime
-from ..loader import ClientConfig
 import json
+
 import grpc
 from google.protobuf import timestamp_pb2
+
+from ..events_pb2 import PushEventRequest, PutLogRequest
+from ..events_pb2_grpc import EventsServiceStub
+from ..loader import ClientConfig
 from ..metadata import get_metadata
 
+
 def new_event(conn, config: ClientConfig):
     return EventClientImpl(
         client=EventsServiceStub(conn),
         token=config.token,
+        namespace=config.namespace or "",
     )
 
+
 def proto_timestamp_now():
     t = datetime.datetime.now().timestamp()
     seconds = int(t)
     nanos = int(t % 1 * 1e9)
 
     return timestamp_pb2.Timestamp(seconds=seconds, nanos=nanos)
 
+
 class EventClientImpl:
-    def __init__(self, client, token):
+    def __init__(self, client, token, namespace):
         self.client = client
         self.token = token
+        self.namespace = namespace
 
     def push(self, event_key, payload):
         try:
-            payload_bytes = json.dumps(payload).encode('utf-8')
+            payload_bytes = json.dumps(payload).encode("utf-8")
         except json.UnicodeEncodeError as e:
             raise ValueError(f"Error encoding payload: {e}")
 
+        event_key = f"{self.namespace}:{event_key}"
```

Comment:
This doesn't seem to match the implementation of the TS SDK: https://github.com/hatchet-dev/hatchet-typescript/blob/eaca1fcfd416665467a6fa8bcc0ac0989dd7512f/src/clients/event/event-client.ts#L33.